Simplify database settings, DB Backup and add new database query forms

JevgenijVisockij opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Simplifies database forms
| Type?         | refacto
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Part of #16482
| How to test?  | Simplifying Advanced Parameters > Database > SQL Manager, Advanced Parameters > Database > SQL Manager > Add new SQL query, Advanced Parameters > Database > DB Backup Forms. Everything must look the same with 2 exceptions: blue help box replaced with help text under input and fields that actually were required before now have red * next to them.

:notebook: **BC Break**
Backwards compatibility break introduced due to extension of TranslationAwareType by SqlRequestSettingsType, SqlRequestType, BackupOptionsType. This means if any module extends any of this types they will get an exception upon upgrading to PS version containing changes in this PR.
<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/21243)
<!-- Reviewable:end -->

JevgenijVisockij committed:
Simplified database settings

JevgenijVisockij committed:
Simplifeid Add SQL request form

JevgenijVisockij committed:
Simplified backup options

JevgenijVisockij committed:
Added required => false to a type which doesn't need it

JevgenijVisockij committed:
CS Fixer fixes

JevgenijVisockij committed:
Removed uneeded required false

Julievrz commented:
Wording looks good to me. 👍

JevgenijVisockij committed:
Modified form so upon saving it renders form again instead of redirecting

atomiix commented on src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Backup/BackupOptionsType.php:
```diff
@@ -26,22 +26,71 @@
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        $configuration
```
```suggestion
        Configuration $configuration
```

JevgenijVisockij committed:
Update src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Backup/BackupOptionsType.php

Added Typehint for Configuration

Co-authored-by: atomiix <tbaccelli@gmail.com>

atomiix commented on src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/RequestSql/Blocks/settings_panel.html.twig:
```diff
@@ -33,19 +34,7 @@
-        {% block request_sql_settings %}
-          {{ form_rest(requestSqlSettingsForm) }}
-        {% endblock %}
+            {{ form_widget(requestSqlSettingsForm) }}
```
```suggestion
        {{ form_widget(requestSqlSettingsForm) }}
```

JevgenijVisockij committed:
Update src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/RequestSql/Blocks/settings_panel.html.twig

Fixed the indetation for html.twig

Co-authored-by: atomiix <tbaccelli@gmail.com>

Progi1984 commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/BackupController.php:
```diff
@@ -156,7 +133,9 @@ public function saveOptionsAction(Request $request)
         }
 
-        return $this->redirectToRoute('admin_backups_index');
+        $backupForm = $this->getBackupFormHandler()->getForm();
```
Really ? But the form is already in `$backupForm`.

JevgenijVisockij replied:
Yeah you are right, this is not needed here, fixed it.

Progi1984 reviewed (commented):
A small feedback

JevgenijVisockij committed:
Removed uneeded   $backupForm = $this->getBackupFormHandler()->getForm();

Progi1984 commented:
But PHPStan is :red_circle:

JevgenijVisockij committed:
Modified sql requests so it renders form upon save instead of redirecting

# Conflicts:
#	src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SqlManagerController.php

JevgenijVisockij committed:
CS fixer fixes

# Conflicts:
#	src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SqlManagerController.php

JevgenijVisockij committed:
Restored backup and sql manager controllers

JevgenijVisockij committed:
Removed uneeded space in backup controller

JevgenijVisockij committed:
Removed uneeded space in sql manager controller

PierreRambaud commented on src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml:
```diff
@@ -754,11 +761,21 @@ services:
+        public: true
+        tags:
+            - { name: form.type }
+
+
```
```suggestion

```

PierreRambaud commented on src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Backup/BackupOptionsType.php:
```diff
@@ -26,22 +26,71 @@
+            '%prefix%connections, %prefix%connections_page %prefix%connections_source, %prefix%guest, %prefix%statssearch'
+        );
+
+        $backupDropTablesHelp = $this->trans(
+            'If enabled, the backup script will drop your tables prior to restoring data.',
+            'Admin.Advparameters.Help'
+        );
+
+        $backupDropTablesHelp .= ' ';
+        $backupDropTablesHelp .= $this->trans('(ie. "DROP TABLE IF EXISTS")', 'Admin.Advparameters.Help');
```
Why not doing a single string?

Julievrz replied:
Well, good question! I thought it was because the example might be different in some cases but if it is not the case, you are right, we could definitely use one string.

Julievrz replied:
@JevgenijVisockij Could you delete this string, please?

JevgenijVisockij replied:
Removed the string.

JevgenijVisockij committed:
Update src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

Removed uneeded line

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

Julievrz commented on src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Backup/BackupOptionsType.php:
```diff
@@ -26,22 +26,71 @@
+        );
+
+        $backupDropTablesHelp = $this->trans(
+            'If enabled, the backup script will drop your tables prior to restoring data.',
```
```suggestion
            'If enabled, the backup script will drop your tables prior to restoring data. (ie. "DROP TABLE IF EXISTS")',
```

JevgenijVisockij committed:
Update src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Backup/BackupOptionsType.php

Added example to full translation of help

Co-authored-by: Julie Varisellaz <70583503+Julievrz@users.noreply.github.com>

JevgenijVisockij committed:
Removed no longer needed string

JevgenijVisockij commented:
https://github.com/PrestaShop/PrestaShop/pull/21652#discussion_r572755026

@Julievrz  Removed.

JevgenijVisockij commented:
This one hasn't been reviewed for a while. Am I missing something that needs to be done?

Julievrz commented:
@Progi1984 Hi, is there something specific you want me to have a look at? The Wording ✔️ label is already there. :)

atomiix commented:
@JevgenijVisockij I don't think this PR is a WIP anymore, right?

Progi1984 commented:
Thanks @JevgenijVisockij & @sarahdib
